### PR TITLE
hook plugin implementation

### PR DESCRIPTION
**Pull request by jaymccon**
## Overview

This PR adds a pluggable way to add hooks that run before or after stacks are launched. Plugins receive test definitions, taskcat config object and rendered parameters, with stack outputs coming in a future pr.

example project config using plugins:

```
tests:
  foobar:
    prehooks:
    - type: cfn-lint
      config:
        strict: true
        custom-resource-spec: ./my-resource-spec.json
    posthooks:
    - type: run-command
      config:
        command: touch blah
        timeout: 100
```

plugins must be installed using the `type` prefixed with `taskcat_plugin_` as package name, and `Hook` as class name. These can be bundled into taskcat when it installs, or installed separately from a plugin owned and published by 3rd parties.

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -329,7 +329,7 @@ indent-after-paren=4
 indent-string='    '
 
 # Maximum number of characters on a single line.
-max-line-length=88
+max-line-length=140
```
Comment (jaymccon) [inline .pylintrc:332]: pre-commit was failing due to other files not changed in this pr due to line length, so I increased it 👀

**Review by andrew-glenn — APPROVED** [review]

**Pull request merged**